lib/gdal/extensions/raster_band/algorithm_extensions.rb

```diff
@@ -0,0 +1,111 @@
+# frozen_string_literal: true
+
+require 'gdal/raster_band'
+
+module GDAL
+  class RasterBand
+    # RasterBand methods for doing warp-like things, but not using GDAL's Warp
+    # API.
+    module AlgorithmExtensions
```

Comment:
Do we use this anywhere??

Reply:
Looks like we're using it in one place to initialize a raster band to NODATA.  There's probably a better way we could be doign that.

Reply:
I think I'll leave this one; seems generic enough to be useful for those that don't wanna use the Warp API.